src/webui/abstractwebapplication.cpp

```diff
@@ -310,10 +310,15 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
+    return ((!env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("10.0.0.0/8"))
+            && !env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("172.16.0.0/12"))
+            && !env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("192.168.0.0/16"))
+            && !env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("fc00::/7")))
+            || Preferences::instance()->isWebUiLANAuthEnabled())
+            && ((env_.clientAddress != QHostAddress::LocalHost
```

Comment:
> Shouldn't be OR in this place?

The LAN conditionals return true if the address isn't localhost OR if local auth is enabled. Therefore if an OR is used and a user connects with a LAN ip, the function would return true regardless of the LAN conditionals. 

That being said, development is still underway, so don't start picking apart my changes just yet ;) 

Once I'm finished I'll squash the changes and update this PR.